File: lab1/selenium_java/CarritoPrueba.java
import java.time.Duration;
import java.util.List;
import java.util.Map;

import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.support.ui.WebDriverWait;

import pages.CarritoPagina;
import pages.InventarioPagina;
import pages.LoginPagina;

public class CarritoPrueba {

    private WebDriver driver;
    private WebDriverWait espera;
    private LoginPagina paginaLogin;
    private InventarioPagina paginaInventario;
    private CarritoPagina paginaCarrito;

    @Before
    public void configurar() {
        ChromeOptions opciones = new ChromeOptions();
        opciones.setExperimentalOption("prefs", Map.of(
            "credentials_enable_service", false,
            "profile.password_manager_enabled", false,
            "profile.password_manager_leak_detection", false
        ));
        opciones.addArguments("--disable-save-password-bubble");
        // si se quiere usar el chromedriver.exe descomentar y comentar lo de abajo
        // System.setProperty("webdriver.chrome.driver", "../drivers/chromedriver.exe");
        // WebDriverManager.chromedriver().setup();
        driver = new ChromeDriver(opciones);
        driver.manage().window().maximize();
        espera = new WebDriverWait(driver, Duration.ofSeconds(5));
        driver.get("https://www.saucedemo.com/");

        paginaLogin      = new LoginPagina(driver);
        paginaInventario = new InventarioPagina(driver);
        paginaCarrito    = new CarritoPagina(driver);

        pausa(1000);
        paginaLogin.iniciarSesion("standard_user", "secret_sauce");
        paginaLogin.esperarInventario();
        pausa(1200);
    }

    @Test
    public void agregarProductosAlCarrito() {
        int cantidad = 3;
        paginaInventario.agregarProductos(cantidad);

        int contador = paginaInventario.obtenerContadorCarrito();

        paginaInventario.irAlCarrito();
        List<String> productos = paginaCarrito.obtenerNombresProductos();

        System.out.println("Productos agregados (" + productos.size() + "):");
        for (String p : productos) System.out.println("  [+] " + p);
        System.out.println("===========================================\n");

        pausa(5000);

        Assert.assertEquals("El carrito debe tener " + cantidad + " productos", cantidad, contador);
    }

    @Test
    public void validarProductosEnCarrito() {
        int cantidad = 2;
        paginaInventario.agregarProductos(cantidad);
        paginaInventario.irAlCarrito();

        List<String> productos = paginaCarrito.obtenerNombresProductos();
        int cantidadEnCarrito  = paginaCarrito.obtenerCantidadProductos();

        System.out.println("Productos en carrito (" + productos.size() + "):");
        for (String p : productos) System.out.println("  [OK] " + p);
        System.out.println("===========================================\n");

        pausa(5000);

        Assert.assertEquals("El carrito debe contener " + cantidad + " items", cantidad, cantidadEnCarrito);
    }

    @After
    public void cerrar() {
        if (driver != null) driver.quit();
    }

    private void pausa(long ms) {
        try { Thread.sleep(ms); } catch (InterruptedException e) { Thread.currentThread().interrupt(); }
    }
}

File: lab1/selenium_python/CarritoPrueba.py
import unittest
import time
from selenium import webdriver
from selenium.webdriver.chrome.options import Options
from paginas.LoginPagina import LoginPagina
from paginas.InventarioPagina import InventarioPagina
from paginas.CarritoPagina import CarritoPagina


class CarritoPrueba(unittest.TestCase):

    def setUp(self):
        opciones = Options()
        opciones.add_experimental_option("prefs", {
            "credentials_enable_service": False,
            "profile.password_manager_enabled": False,
            "profile.password_manager_leak_detection": False
        })
        opciones.add_argument("--disable-save-password-bubble")

        # Para usar chromedriver.exe manual, descomentar y comentar la linea de abajo:
        # from selenium.webdriver.chrome.service import Service
        # self.driver = webdriver.Chrome(service=Service("../../drivers/chromedriver.exe"), options=opciones)
        self.driver = webdriver.Chrome(options=opciones)
        self.driver.maximize_window()
        self.driver.get("https://www.saucedemo.com/")

        self.pagina_login      = LoginPagina(self.driver)
        self.pagina_inventario = InventarioPagina(self.driver)
        self.pagina_carrito    = CarritoPagina(self.driver)

        time.sleep(1.0)
        self.pagina_login.iniciar_sesion("standard_user", "secret_sauce")
        self.pagina_login.esperar_inventario()
        time.sleep(1.2)

    def test_agregar_productos_al_carrito(self):
        cantidad = 3
        self.pagina_inventario.agregar_productos(cantidad)

        contador = self.pagina_inventario.obtener_contador_carrito()

        self.pagina_inventario.ir_al_carrito()
        productos = self.pagina_carrito.obtener_nombres_productos()

        print(f"Productos agregados ({len(productos)}):")
        for p in productos:
            print(f"  [+] {p}")
        print("===================================================\n")

        time.sleep(5)

        self.assertEqual(cantidad, contador, f"El carrito debe tener {cantidad} productos")

    def test_validar_productos_en_carrito(self):
        cantidad = 2
        self.pagina_inventario.agregar_productos(cantidad)
        self.pagina_inventario.ir_al_carrito()

        productos       = self.pagina_carrito.obtener_nombres_productos()
        cantidad_carrito = self.pagina_carrito.obtener_cantidad_productos()

        print(f"Productos en carrito ({len(productos)}):")
        for p in productos:
            print(f"  [OK] {p}")
        print("===================================================\n")

        time.sleep(5)

        self.assertEqual(cantidad, cantidad_carrito, f"El carrito debe contener {cantidad} items")

    def tearDown(self):
        if self.driver:
            self.driver.quit()
